### PR TITLE
Add Gpu timing native rendering plugin

### PR DESCRIPTION
**Pull request by danandersson**
Add Gpu timing native rendering plugin to insert and fetch timing requests in the command queue.
This plugin will be used for upcoming holotoolkit unity contributions

### CODE REVIEW

**Comment by StephenHodgson** [comment]
OoooOooo this looks cool!  Thanks!

File: GpuTiming/GpuTiming/GpuTiming.cpp [hunk 1]
```diff
@@ -0,0 +1,344 @@
+﻿#include <pch.h>
```
Comment (NeerajW) [inline GpuTiming/GpuTiming/GpuTiming.cpp]: Files need copyright headers.

File: GpuTiming/GpuTiming/GpuTiming.cpp [hunk 2]
```diff
@@ -0,0 +1,344 @@
...
+		s_Device->CreateQuery(&disjointDesc, &mDisjoint);
+		s_Device->CreateQuery(&desc, &mBegin);
+		s_Device->CreateQuery(&desc, &mEnd);
+	}
+
+	void DestroyEventQueries()
+	{
+		if (mDisjoint == nullptr)
+		{
+			mDisjoint->Release();
```
Comment (NeerajW) [inline GpuTiming/GpuTiming/GpuTiming.cpp]: Should these be ComPtr or auto may be? Asking as I don't know enough here.
That way you might be able to skip manually managing lifetime and references?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,4 @@
+.vs/
```
Comment (NeerajW) [inline .gitignore]: Please add a ReadME. Sorry if I missed it.

**Pull request merged**